dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlanner.cs

```diff
@@ -46,8 +47,13 @@ public Task<HandlebarsPlan> CreatePlanAsync(Kernel kernel, string goal, Cancella
     {
         Verify.NotNullOrWhiteSpace(goal);
 
-        // TODO (@teresaqhoang): Add instrumentation without depending on planners.core
-        return this.CreatePlanCoreAsync(kernel, goal, cancellationToken);
+        var logger = kernel.GetService<ILoggerFactory>().CreateLogger(this.GetType());
```

Comment:
No need for run-time reflection - Compile-time type determination is possible:

```suggestion
        var logger = kernel.GetService<ILoggerFactory>().CreateLogger(typeof(HandlebarsPlanner));
```